utils/get_test_reports.py

```diff
@@ -0,0 +1,247 @@
+"""
+This util provides a way to manually run the tests of the transformers repo as they would be run by the CI.
+It was mainly used for models tests, so if you find features missing for another suite, do not hesitate to open a PR.
+
+Functionnalities:
+- Running specific test suite (models, tokenizers, etc.)
+- Parallel execution across multiple processes (each has to be launched separately with different `--processes` argument)
+- GPU/CPU test filtering and slow tests filter
+- Temporary cache management for isolated test runs
+- Resume functionality for interrupted test runs
+- Important models subset testing
+
+Example usages are below.
+"""
+
+import argparse
+import contextlib
+import os
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Optional
+
+import torch
+
+from .important_files import IMPORTANT_MODELS
+
+
+def is_valid_test_dir(path: Path) -> bool:
+    """Check if a given path represents a valid test dir: the path must point to a dir, not start with '__' or '.'"""
+    return path.is_dir() and not path.name.startswith("__") and not path.name.startswith(".")
+
+
+def run_pytest(
+    suite: str, subdir: Path, root_test_dir: Path, machine_type: str, dry_run: bool, tmp_cache: str, cpu_tests: bool
+) -> None:
+    """
+    Execute pytest on a specific test directory with configured options:
+        - suite (str): name of the test suite being run (e.g., 'models', 'tokenizers')
+        - subdir (Path): the specific directory containing tests to run
+        - root_test_dir (Path): the root directory of all tests, used for relative paths
+        - machine_type (str): type of machine/environment (e.g., 'cpu', 'single-gpu', 'multi-gpu')
+        - dry_run (bool): if True, only print the command without executing it
+        - tmp_cache (str): prefix for temporary cache directory. If empty, no temp cache is used
+        - cpu_tests (bool): if True, include CPU-only tests; if False, exclude non-device tests
+    """
+    relative_path = subdir.relative_to(root_test_dir)
+    report_name = f"{machine_type}_{suite}_{relative_path}_test_reports"
```

Comment:
just want to point out this might not correspond exactly to the artifact names we have when running with Github Actions. 

But if you are OK with this , fine for me.

Reply:
I am OK with it but I can correct it if you want! Could you explain the change if it is easy to do so please?